### DISCUSSION TITLE
Calculate height size dynamicaly

### DISCUSSION THREAD

**Discussion by sebastienL** [scenee/FloatingPanel#469] [Q&A]
I have a position 'tip' of 'FloatingPanelPosition".
is it possible to calculate the height automatically? 
or do you have to calculate the height manually?

what I will :
`func insetFor(position: FloatingPanelPosition) -> CGFloat? {
        switch position {
        case .tip: return 100
        default: return nil
        }
}`

Thanks

**Comment by scenee** [comment]
> is it possible to calculate the height automatically? or do you have to calculate the height manually?

Could you please give me examples for each automatic and manual height?

**Reply by sebastienL** [reply]
Thanks, @scenee 

I managed to calculate the height of the bottom sheet with `FloatingPanelAdaptiveLayoutAnchor.`

Do we have to put my view in a scrollview, as in the example: `Show Adaptive Panel`? (see picture)

If the bottom sheet is bigger than the screen, with `FloatingPanelAdaptiveLayoutAnchor`, can we give it a maximum height? 
to choose between a fixed size and an adaptive size? (see picture)

![Simulator Screen Shot - iPhone 12 Pro - 2021-05-26 at 16 23 20](https://user-images.githubusercontent.com/3176155/119677783-4da7ed00-be3f-11eb-8a22-03b507cffe56.png)

![Capture d’écran 2021-05-26 à 16 22 36](https://user-images.githubusercontent.com/3176155/119677886-60222680-be3f-11eb-8371-0bbc14eb7baf.png)

**Comment by sebastienL** [comment]
@scenee  I succeeded. :)
on the other hand, how to make so that the bottom Sheet is blocked at a certain height of the screen.

if I put an offet at 300 with the method
`.full: FloatingPanelAdaptiveLayoutAnchor (absoluteOffset: 300,
contentLayout: targetGuide,
referenceGuide: .safeArea) `

it's OK.

but this offset works even when the items in the bottomSheet are small.

Is it possible to set the offset to 0 when the bottomSheet does not reach a certain height and put one when it exceeds a certain height of the screen?

**Reply by scenee** [reply]
Thanks for your effort and trying v2.0 APIs. 

> Is it possible to set the offset to 0 when the bottomSheet does not reach a certain height and put one when it exceeds a certain height of the screen?

I’m sorry it's not possible to do that. But I woud like to consider that to enhance `FloatingPanelAdaptiveLayoutAnchor` API for your use case because the opt-in functionality would also resolve https://github.com/scenee/FloatingPanel/issues/413.

**Reply by scenee** [reply]
The updated API maybe looks like this.

```diff
-    @objc public init(absoluteOffset offset: CGFloat, contentLayout: UILayoutGuide, referenceGuide: FloatingPanelLayoutReferenceGuide = .safeArea) {
+    @objc public init(absoluteOffset offset: CGFloat, contentLayout: UILayoutGuide, referenceGuide: FloatingPanelLayoutReferenceGuide = .safeArea, maxBounds: FloatingPanelLayoutReferenceGuide? = nil) {
```

**Reply by sebastienL** [reply]
Hey, 

I’m sorry for the late response.
I don't see your proposition in the last version ?.

I can test your solution if necessary

Thx

> The updated API maybe looks like this.
> 
> ```diff
> -    @objc public init(absoluteOffset offset: CGFloat, contentLayout: UILayoutGuide, referenceGuide: FloatingPanelLayoutReferenceGuide = .safeArea) {
> +    @objc public init(absoluteOffset offset: CGFloat, contentLayout: UILayoutGuide, referenceGuide: FloatingPanelLayoutReferenceGuide = .safeArea, maxBounds: FloatingPanelLayoutReferenceGuide? = nil) {
> ```

**Reply by scenee** [reply]
I'm sorry I has not yet implemented, but I've done the initial implementation. Now I'm testing it. So I'm going to create a PR in a few week 😉

**Reply by AlvaroFranco** [reply]
Would be great to have this feature implemented. Looking into it, great work @scenee !

**Reply by scenee** [reply]
Hi all. I've finally made it.  Here is the PR to address this request. Thanks for your patient.

**Comment by scenee** [comment]
Thank you for your patient. I support this feature in the `master` branch. This will be released in v2.6.0.
